snap_plugin/v1/tests/test_stream_collector.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# http://www.apache.org/licenses/LICENSE-2.0.txt
+#
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import sys
+import time
+
+import grpc
+
+import snap_plugin.v1 as snap
+from snap_plugin.v1.collect_arg import CollectArg
+from snap_plugin.v1.plugin_pb2 import StreamCollectorStub, Empty
+from snap_plugin.v1.tests import ThreadPrinter
+
+from .mock_plugins import MockStreamCollector
+
+
+def test_monitor():
+    sys.argv = ["", '{"LogLevel": 1, "PingTimeoutDuration": 100}']
+    col = MockStreamCollector("MyStreamCollector", 1)
+    # with a PingTimeoutDuration at 50ms the plugin should shutdown
+    # in just over 150ms
+    col.start()
+    assert col._shutting_down is False
+    time.sleep(.4)
+    assert col._shutting_down is True
+
+
+def test_stream():
+    sys.stdout = ThreadPrinter()
```

Comment:
Testing `StreamMetrics`, `GetMetricTypes` and `GetConfigPolicy` could be all done using one client shared between tests by `@pytest.fixture`, which would speed up the tests (no need to start GRPC server 3 times)